src/main/java/org/reaktivity/nukleus/sse/internal/stream/ServerStreamFactory.java

```diff
@@ -801,36 +801,52 @@ private void handleReset(
         private void handleSignal(
             SignalFW signal)
         {
-            final HttpSignalExFW httpSignalEx = httpSignalExRO.tryWrap(signal.buffer(),
-                                                                       signal.extension().offset(),
-                                                                       signal.extension().limit());
-            if (httpSignalEx != null && signal.extension().sizeof() == httpSignalExRO.limit())
+            final HttpSignalExFW httpSignalEx = signal.extension().get(httpSignalExRO::tryWrap);
+            if (httpSignalEx != null)
             {
                 final ListFW<HttpHeaderFW> httpHeaders = httpSignalEx.headers();
-                final Map<String, String> collectedHeaders = new HashMap<>();
-                httpHeaders.forEach(header -> collectedHeaders.put(header.name().asString(), header.value().asString()));
+                final Map<String, String> challengeHeaders = new HashMap<>();
+                httpHeaders.forEach(header -> challengeHeaders.put(header.name().asString(), header.value().asString()));
```

Comment:
Can we remove `challengeHeaders`, and use `ListFW<HttpHeaderFW>.forEach(...)` below instead of `Map<String, String>.forEach(...)` ?